contribute/design/1.x/maya/20190107-cStor-volume-resize.md

```diff
@@ -0,0 +1,259 @@
+---
+oep-number: draft Resize 20190701
+title: cStor Volume Resize
+authors:
+  - "@mittachaitu"
+owners:
+  - "@kmova"
+  - "@vishnuitta"
+  - "@amitkumardas"
+  - "@payes"
+  - "@pawanpraka1"
+editor: "@mittachaitu"
+creation-date: 2019-07-01
+last-updated: 2019-07-01
+status: provisional
+see-also:
+  - NA
+replaces:
+  - NA
+superseded-by:
+  - NA
+---
+
+# cStor Volume Resize
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+* [Proposal](#proposal)
+    * [User Stories](#user-stories)
+      * [Resize cStor Volume](#resize-cstor-volume)
+    * [Implementation Details](#implementation-details)
+    * [Custom Resources used to resize cStor volume](#custom-resource-used-to-resize)
+    * [Current Implementation](#current-implementation)
+      * [High Level cStor Volume Resize Workflow](#high-level-cStor-volume-resize-workflow)
+    * [High Level Design](#high-level-design)
+      * [CStorVolumeClaim -- existing custom resource](#cstorvolumeclaim----existing-custom-resource)
+      * [CStorVolume -- existing custom resource](#cstorvolume----existing-custom-resource)
+      * [CStorVolumeReplica -- existing custom resource](#cstorvolumereplica----existing-custom-resource)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+
+## Summary
+
+This proposal charts out the design details to implement resize workflow on cStor Volumes.
+
+## Motivation
+
+Resizing volume is the most prominent feature for any storage providers.
+In this we will concentrate only on cStor volumes.
+
+### Goals
+
+- Ability to resize a cStor volume in kubernetes cluster by taking trigger
+  point as pvc spec capacity change.
+
+## Proposal
+
+### User Stories
+
+#### Resize cStor Volume:
+As an application developer I should be able to resize a volume on
```

Comment:
on the fly means that it can be resized while the application pods is still running. If this doesn't support online resizing, we have explicitly mention it.

Reply:
We are supporting online as well as offline.